ArjunaCore/arjuna/tests/classes/com/hp/mwtests/ts/arjuna/atomicaction/AtomicActionAsyncTest.java

```diff
@@ -49,6 +56,28 @@ public void testAbort() throws Exception {
     }
 
     @Test
+    public void testAbortAsyncAC() throws Exception {
+        AtomicActionTestBase.init(false);
+        arjPropertyManager.getCoordinatorEnvironmentBean().setAsyncAfterSynchronization(true);
+
+        final Thread thread = Thread.currentThread();
+        executeTest(true, ActionStatus.COMMITTED, new SyncRecord[] { new SyncRecord() {
+            @Override
+            public boolean afterCompletion(int status) {
+                testAbortAsyncAC = testAbortAsyncAC && (Thread.currentThread() != thread);
```

Comment:
It means that if someone were to rework the implementation he would need to puzzle over why an existing test is failing. Can you add more commentary about why asserting that the callbacks run in different threads from the caller make the exception disappear.